### DISCUSSION TITLE
Triggering Keyboard OS shortcuts from a UGS UI button

### DISCUSSION THREAD

**Discussion by MadTooler** [winder/Universal-G-Code-Sender#2113] [Help]
Is there a way to create a macro or toolbox function that can be assigned to a UI button and can issue a keyboard shortcut command to the OS?

What I am trying to do is send an open window command to my rpi OS to switch from UGS to another running program.  Both programs will be running in full screen mode.  Rpi takes keyboard Alt+Tab to do this.  From the CNC, the only user interface is a touchscreen, so a button will have to be added somewhere in the UGS UI.  I am sure there is a more precise command to also do this within the OS, but I figured/assumed a keyboard shortcut might be closer to achievable within the UGS limitations.

Along with the ability to add this to a button in the UI, I am also trying to have UGS open in full screen mode.  Is that possible?

**Comment by breiler** [comment]
No that is not possible unfortunately.

**Reply by MadTooler** [reply]
Thanks for the quick reply.  Is that a no also for UGS open in full screen?

**Reply by MadTooler** [reply]
Please let me know if UGS can open in full screen, when you get the chance.

For now, since I have NodeRed running on my system as well, I have created a couple of nodes that send hotkeys to the system to achieve my needs of switching windows and loading full screen.  It is delicate since it is dependent on timing of when things actually load and that there are no other programs or windows that may popup, but it does give me the functionality I was attempting.  As for the gui button, I have a series of pin actions passing between my grblhal pico and a custom pico based motor driver.  From there, I am able to cobble together a sequence from UGS macro button pin toggles to determine it is time to send a serial command to my node red to switch screens.  Lots of cogs in the worx, but it does work until it blows up in my face:)
